flocker/node/_deploy.py

```diff
@@ -530,137 +570,132 @@ def calculate_necessary_state_changes(self, desired_state,
+            desired_node_applications
+        ))
+        for application_name in applications_to_inspect:
+            inspect_desired = desired_applications_dict[application_name]
+            inspect_current = current_applications_dict[application_name]
+            if inspect_desired != inspect_current:
+                changes = [
+                    StopApplication(application=inspect_current),
+                    StartApplication(application=inspect_desired,
+                                     hostname=self.hostname)
+                ]
+                sequence = Sequentially(changes=changes)
+                if sequence not in restart_containers:
+                    restart_containers.append(sequence)
+
+        # Find any dataset that are moving to or from this node - or
+        # that are being newly created by this new configuration.
+        dataset_changes = find_dataset_changes(
+            self.hostname, current_cluster_state, desired_configuration)
+
+        if dataset_changes.resizing:
+            phases.append(InParallel(changes=[
+                ResizeDataset(dataset=dataset)
+                for dataset in dataset_changes.resizing]))
+
+        # Do an initial push of all volumes that are going to move, so
+        # that the final push which happens during handoff is a quick
+        # incremental push. This should significantly reduces the
+        # application downtime caused by the time it takes to copy
+        # data.
+        if dataset_changes.going:
+            phases.append(InParallel(changes=[
+                PushDataset(dataset=handoff.dataset,
+                            hostname=handoff.hostname)
+                for handoff in dataset_changes.going]))
+
+        if stop_containers:
+            phases.append(InParallel(changes=stop_containers))
+        if dataset_changes.going:
+            phases.append(InParallel(changes=[
+                HandoffDataset(dataset=handoff.dataset,
+                               hostname=handoff.hostname)
+                for handoff in dataset_changes.going]))
+        # any datasets coming to this node should also be
+        # resized to the appropriate quota max size once they
+        # have been received
+        if dataset_changes.coming:
+            phases.append(InParallel(changes=[
+                WaitForDataset(dataset=dataset)
+                for dataset in dataset_changes.coming]))
+            phases.append(InParallel(changes=[
+                ResizeDataset(dataset=dataset)
+                for dataset in dataset_changes.coming]))
+        if dataset_changes.creating:
+            phases.append(InParallel(changes=[
+                CreateDataset(dataset=dataset)
+                for dataset in dataset_changes.creating]))
+        start_restart = start_containers + restart_containers
+        if start_restart:
+            phases.append(InParallel(changes=start_restart))
+        return Sequentially(changes=phases)
```

Comment:
Pretty much. Other change was referring to `self.hostname` instead of passed in one, and using passed in local state. But no significant changes (which is why all the unit tests changes were trivial).